### PR TITLE
Achievements

### PR DESCRIPTION
**Pull request by Amelia0010**
### Intent of your Pull Request

Converts the medals system into achievements, and moves them to the database. I removed all of the achievements except for the bubblegum one (since that actually does something), but I'll add more at some point.

To do list:

- [x] Make a UI, possibly adapting the one from this PR: https://github.com/tgstation/tgstation/pull/47058
- [x] Add the necessary schema changes, since they're currently not there

#### Changelog

:cl:  
experimental: Medals have been converted into achievements and moved from the hub to the database. This allows us to enable them.
/:cl:

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
We have medals turned off tho, so even the bubble one doesn’t do anything

**Comment by Amelia0010** [comment]
>experimental: Medals have been converted into achievements and moved from the hub to the database. This allows us to enable them.

**Comment by MegaEmpirical** [comment]
When did we turn medals off? I know they're not used often but I thought they were cool.

**Comment by Amelia0010** [comment]
They were never turnt on

**Comment by MegaEmpirical** [comment]
Hmm, I could have swore I've seen them at round end when they've been distributed.

**Comment by HomocidlePancakes** [comment]
>Hmm, I could have swore I've seen them at round end when they've been distributed.

Wrong kind of meals m8

File: code/datums/achievements/viewer.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//UI adapted from https://github.com/tgstation/tgstation/pull/47058
+
+/datum/achievement_browser
+	var/client/client
+	var/achievements = list()
```
Comment (AshCorr) [inline code/datums/achievements/viewer.dm]: This variable isn't really being used to its full extent, probably worth making it a local variable just in the methods that use it.

**Review by AshCorr — DISMISSED** [review]

File: code/controllers/subsystem/achievements.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	qdel(ridOldChieves)
+	return ..()
+
+/datum/controller/subsystem/achievements/proc/unlock_achievement(datum/achievement/achievement, client/C)
+	if(!achievements[achievement])
+		stack_trace("Achievement [initial(achievement.name)] not found in list of achievements when trying to unlock for [C.ckey]")
+		return FALSE
+	if(!has_achievement(achievement, C))
+		var/datum/DBQuery/medalQuery = SSdbcore.NewQuery("INSERT INTO [format_table_name("earned_achievements")] (ckey, id) VALUES ('[C.ckey]', '[achievements[achievement]]')")
+		medalQuery.Execute()
```
Comment (nichlas0010) [inline code/controllers/subsystem/achievements.dm]: [13:31:13] Runtime in achievements.dm, line 55: Cannot execute 0.Execute().
proc name: unlock achievement (/datum/controller/subsystem/achievements/proc/unlock_achievement)
usr: Nichlas0010/(Maggie May)
usr.loc: (Lavaland Wastes (102, 10, 5))
src: Achievements (/datum/controller/subsystem/achievements)
call stack:
Achievements (/datum/controller/subsystem/achievements): unlock achievement(/datum/achievement/bubblegum (/datum/achievement/bubblegum), Nichlas0010 (/client))
world: WrapAdminProcCall(Achievements (/datum/controller/subsystem/achievements), "unlock_achievement", /list (/list))
WrapAdminProcCall(Achievements (/datum/controller/subsystem/achievements), "unlock_achievement", /list (/list))
Nichlas0010 (/client): Atom ProcCall(Achievements (/datum/controller/subsystem/achievements))

**Pull request merged**